### DISCUSSION TITLE
Requires code_teams and package_protections setup

### DISCUSSION THREAD

**Discussion by oskarpearson** [rubyatscale/pack_stats#11] [General]
Hi All

I really like the idea of this gem (since I've previously implemented something pretty similar!)

On attempting to use it, I've found that I need to configure the following to get useable output:

- code_ownership
- code_teams
- package_protections

Is the intention that this gem could be used independently of the other gems without all of them being configured? If so, I guess I should raise issues for the above?

The code_ownership and code_teams dependencies are not necessarily a problem, though a little unexpected since not covered in the docs. (I needed to add `code_ownership.yml` and `config/teams` files to get things to work) Ideally I'd like to use the data from the package.yml files

However `package_protections` is a bit more wide-ranging. I would like to be able to use this gem without configuring all packages to have the relevant metadata, since I've not investigated that yet and got the whole team to buy into it.

Thanks again for the gem!

**Comment by alexevanczuk** [comment]
Hey @oskarpearson thanks for filing this. It's definitely the intention that this gem can be used gradually and without having fully configured all other gems. As you pointed out, it was a bit simpler with code ownership and code teams, but package protections is a bit more of an investment and change to developer UX.

Can you share what error message you're receiving? I've actually been moving `package_protections` functionality to https://github.com/rubyatscale/rubocop-packs and trying to move towards a pattern of having pack-specific `.rubocop.yml` files instead of the `package.yml` metadata. In the meantime, I'm wondering if I can just loosen the constraints in `package_protections` so that it works without having expected you to have configured everything "correctly."

Also note that right now this gem works by looking at the metadata in `package.yml`, so we'd need to change this to look at `enforce_privacy` and `enforce_dependencies` so it produces correct results for people not using package protections.

**Comment by alexevanczuk** [comment]
Hey @oskarpearson I just pushed up some PRs to `package_protections` and `modularization_statistics` that I *think* should allow you to get some value out of this gem without configuring `package_protections`. Give it a try and let me know what you think. The version is 1.38.0.

**Comment by oskarpearson** [comment]
Thanks so much for taking this on so quickly! Really appreciate it.

I'll do some experimentation and raise any issues/errors I find in issues or PRs

**Comment by oskarpearson** [comment]
@alexevanczuk The updates that you've done worked for me

Your comment above refers to 1.38.0, but that doesn't appear at https://rubygems.org/gems/modularization_statistics so I pointed at the git repo's master for my testing. There's no tag for it at https://github.com/rubyatscale/modularization_statistics/tags either. Did you mean to do a release? Or were you wanting feedback before doing so?

I added a github issue for some improvements https://github.com/rubyatscale/modularization_statistics/issues/14 - am happy to submit a PR if you think the direction there is right

Thanks again for the gem and for your help!

**Comment by alexevanczuk** [comment]
Whoops @oskarpearson – CD had failed. I fixed the issue and it should be out now.

I'll check out your issue! Thank you!
